syncd/scripts/syncd_init_common.sh

```diff
@@ -24,17 +24,15 @@ fi
 # Use temporary view between init and apply
 CMD_ARGS+=" -u"
 
-BOOT_TYPE="$(cat /proc/cmdline | grep -o 'SONIC_BOOT_TYPE=\S*' | cut -d'=' -f2)"
-
-case "$BOOT_TYPE" in
-  fast-reboot)
-     FAST_REBOOT='yes'
-    ;;
-  fastfast)
+case "$(cat /proc/cmdline)" in
+  *fastfast*)
     if [ -e /var/warmboot/warm-starting ]; then
         FASTFAST_REBOOT='yes'
     fi
     ;;
+  *fast*)
```

Comment:
>fast [](start = 3, length = 4)

This one is also too general. We can match for 2 patterns:

1. `*SONIC_BOOT_TYPE=fast*`
2. `*fast-reboot*`